### PR TITLE
Client Incentives: require DAO permission to withdraw, refactor client state for better NFT art

### PR DESCRIPTION
**Pull request by eladmallel**

### CODE REVIEW

File: packages/nouns-contracts/contracts/client-incentives/INounsClientTokenTypes.sol [hunk 1]
```diff
@@ -17,7 +17,17 @@ pragma solidity ^0.8.19;
 
 interface INounsClientTokenTypes {
     struct ClientMetadata {
+        /// @notice Whether the DAO has approved the client to withdraw their rewards.
+        bool approved;
+        /// @notice The amount of reward tokens this client has been rewarded.
+        uint104 rewarded;
```
Comment (davidbrai) [inline packages/nouns-contracts/contracts/client-incentives/INounsClientTokenTypes.sol:23]: @eladmallel any particular reason you decided on uint104?

File: packages/nouns-contracts/contracts/client-incentives/Rewards.sol [hunk 2]
```diff
@@ -165,20 +185,40 @@ contract Rewards is NounsClientToken, UUPSUpgradeable, PausableUpgradeable {
...
-    function registerClient(
-        string calldata name,
-        string calldata description
-    ) public override whenNotPaused returns (uint32) {
+    function registerClient(string calldata name, string calldata description) external whenNotPaused returns (uint32) {
         RewardsStorage storage $ = _getRewardsStorage();
 
-        uint32 tokenId = super.registerClient(name, description);
+        uint32 tokenId = $.nextTokenId;
+        $.nextTokenId++;
```
Comment (davidbrai) [inline packages/nouns-contracts/contracts/client-incentives/Rewards.sol:192]: @eladmallel nit: maybe more gas efficient to do `$.nextTokenId = tokenId + 1` here

File: packages/nouns-contracts/contracts/client-incentives/Rewards.sol [hunk 3]
```diff
@@ -542,6 +601,15 @@ contract Rewards is NounsClientToken, UUPSUpgradeable, PausableUpgradeable {
      * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
      */
 
+    /**
+     * @dev Only `owner` can call this function
```
Comment (davidbrai) [inline packages/nouns-contracts/contracts/client-incentives/Rewards.sol:605]: @eladmallel worth adding a `@notice` explaining what's the significance of this flag

File: packages/nouns-contracts/contracts/client-incentives/Rewards.sol (hunk 2 above)
Comment (eladmallel) [reply packages/nouns-contracts/contracts/client-incentives/Rewards.sol]: fixed

File: packages/nouns-contracts/contracts/client-incentives/Rewards.sol (hunk 3 above)
Comment (eladmallel) [reply packages/nouns-contracts/contracts/client-incentives/Rewards.sol:608]: added

**Pull request merged**